src/pkg/packager/mirror.go

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package packager contains functions for interacting with, managing and deploying Zarf packages.
+package packager
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/helpers"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// Mirror pulls resources from a package (images, git repositories, etc) and pushes them to remotes in the air gap without deploying them
+func (p *Packager) Mirror(noImgChecksum bool) (err error) {
+	spinner := message.NewProgressSpinner("Mirroring zarf package %s", p.cfg.PkgOpts.PackagePath)
+	defer spinner.Stop()
+
+	if helpers.IsOCIURL(p.cfg.PkgOpts.PackagePath) {
+		err := p.SetOCIRemote(p.cfg.PkgOpts.PackagePath)
+		if err != nil {
+			return err
+		}
+	}
+
+	if err := p.loadZarfPkg(); err != nil {
+		return fmt.Errorf("unable to load the Zarf Package: %w", err)
+	}
+
+	if err := ValidatePackageSignature(p.tmp.Base, p.cfg.PkgOpts.PublicKeyPath); err != nil {
+		return err
+	}
+
+	// Confirm the overall package mirror
+	if !p.confirmAction(config.ZarfMirrorStage, p.cfg.SBOMViewFiles) {
+		return fmt.Errorf("mirror cancelled")
+	}
+
+	state := &types.ZarfState{
+		RegistryInfo:   p.cfg.InitOpts.RegistryInfo,
+		GitServer:      p.cfg.InitOpts.GitServer,
+		ArtifactServer: p.cfg.InitOpts.ArtifactServer,
+	}
+	p.cfg.State = state
+
+	// Filter out components that are not compatible with this system if we have loaded from a tarball
+	p.filterComponents(true)
+	requestedComponentNames := getRequestedComponentList(p.cfg.PkgOpts.OptionalComponents)
+
+	for _, component := range p.cfg.Pkg.Components {
+		if len(requestedComponentNames) == 0 || helpers.SliceContains(requestedComponentNames, component.Name) {
+			if err := p.mirrorComponent(component, noImgChecksum); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// mirrorComponent mirrors a Zarf Component.
+func (p *Packager) mirrorComponent(component types.ZarfComponent, noImgChecksum bool) error {
+	message.Debugf("packager.mirrorComponent(%#v, %#v", p.tmp, component)
+
+	componentPath, err := p.createOrGetComponentPaths(component)
+	if err != nil {
+		return fmt.Errorf("unable to create the component paths: %w", err)
+	}
+
+	// All components now require a name
+	message.HeaderInfof("📦 %s COMPONENT", strings.ToUpper(component.Name))
+
+	hasImages := len(component.Images) > 0
+	hasRepos := len(component.Repos) > 0
+
+	if hasImages {
+		if err := p.pushImagesToRegistry(component.Images, noImgChecksum); err != nil {
+			return fmt.Errorf("unable to push images to the registry: %w", err)
+		}
+	}
+
+	if hasRepos {
+		if err = p.pushReposToRepository(componentPath.Repos, component.Repos); err != nil {
+			return fmt.Errorf("unable to push the repos to the repository: %w", err)
+		}
+	}
```

Comment:
this is where I would put debug statements if I wanted them, as all of the previous operations in this fn are near guaranteed to not fail.

Reply:
I think it is fine to not add debug statements - removed them from the other places (was mostly copy pasta at that point)